src/writeData/subscriptions/components/SubscriptionDetailsPage.tsx

```diff
@@ -0,0 +1,203 @@
+  ParsingForm = 'parsing',
+}
+
+const navigationSteps: SubscriptionNavigationModel[] = [
+  {
+    glyph: IconFont.UploadOutline,
+    name: 'Connect \n to Broker',
+    type: Steps.BrokerForm,
+  },
+  {
+    glyph: IconFont.Subscribe,
+    name: 'Subscribe \n to Topic',
+    type: Steps.SubscriptionForm,
+  },
+  {
+    glyph: IconFont.Braces,
+    name: 'Define Data \n Parsing Rules',
+    type: Steps.ParsingForm,
+  },
+]
+
+const SubscriptionDetailsPage: FC = () => {
+  const [active, setFormActive] = useState<Steps>(Steps.BrokerForm)
+  const {currentSubscription, loading, saveForm, updateForm} = useContext(
+    SubscriptionUpdateContext
+  )
+  const {change} = useContext(SubscriptionListContext)
+  const showUpgradeButton = useSelector(shouldShowUpgradeButton)
+  const {id} = useParams<{id: string}>()
+  const buckets = useSelector((state: AppState) =>
+    getAll<Bucket>(state, ResourceType.Buckets).filter(b => b.type === 'user')
+  )
+  const {bucket} = useContext(WriteDataDetailsContext)
+  useEffect(() => {
+    change(id)
+  }, [id, change])
+  const [edit, setEdit] = useState(false)
+
+  const handleClick = (step: number) => {
+    setFormActive(navigationSteps[step - 1].type as Steps)
+  }
+
+  const getActiveStep = activeForm => {
+    let currentStep = 1
+    navigationSteps.forEach((step, index) => {
+      if (step.type === activeForm) {
+        currentStep = index + 1
+      }
+    })
+    return currentStep
+  }
+  return (
+    <GetResources resources={[ResourceType.Buckets]}>
+      <Page>
+        <SpinnerContainer
+          spinnerComponent={<TechnoSpinner />}
+          loading={loading}
+        >
+          <Page.Contents
+            fullWidth={true}
+            scrollable={true}
+            className="subscription-details-page"
+          >
+            {showUpgradeButton && (
+              <FlexBox
+                justifyContent={JustifyContent.FlexEnd}
+                alignItems={AlignItems.FlexEnd}
+                stretchToFitHeight={true}
+              >
+                <CloudUpgradeButton />
+              </FlexBox>
+            )}
+            <div className="subscription-details-page__progress">
+              <SubwayNav
+                currentStep={getActiveStep(active)}
+                onStepClick={handleClick}
+                navigationSteps={navigationSteps}
+                settingUpIcon={FormLogo}
+                settingUpText="MQTT Connector"
+              />
+            </div>
+            <Heading
+              element={HeadingElement.H3}
+              weight={FontWeight.Regular}
+              className="subscription-details-page__status"
+            >
+              Status:
+              <span
+                className={
+                  currentSubscription &&
+                  `subscription-details-page__status--${currentSubscription.status}`
+                }
+              >
+                {currentSubscription && currentSubscription.status}
```

Comment:
currently just returning the nifi status, will check the status copy changes before the next iteration